lib/SqlCall.js

```diff
@@ -16,323 +16,242 @@
-          + iXml.iXmlNodeSqlParmClose();
-        } else {
-          this.xml += iXml.iXmlNodeSqlParmOpen() + params[i][0] + iXml.iXmlNodeSqlParmClose();
-        }
+        this.xml += `<parm${params[i].length ? ` io='${params[i][1]}'` : ''}>${params[i][0]}</parm>`;
       }
     }
-    this.xml += iXml.iXmlNodeSqlExecuteClose();
+    this.xml += '</execute>';
   }
 
   /**
    * @description adds sql table XML
    * @param {array} params
    * @param {object} [options]
    */
-  tables(params, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlTablesOpen(options.error);
-    } else {
-      this.xml += iXml.iXmlNodeSqlTablesOpen();
-    }
-
+  tables(params, options = {}) {
+    this.xml += `<tables${options.error ? ` error='${options.error}'` : ''}>`;
     for (let i = 0; i < params.length; i += 1) {
-      this.xml += iXml.iXmlNodeSqlParmOpen() + params[i] + iXml.iXmlNodeSqlParmClose();
+      this.xml += `<parm>${params[i]}</parm>`;
     }
-    this.xml += iXml.iXmlNodeSqlTablesClose();
+    this.xml += '</tables>';
   }
 
   /**
    * @description adds sql table priv XML
    * @param {array} params
    * @param {object} [options]
    */
-  tablePriv(params, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlTableprivOpen(options.error);
-    } else {
-      this.xml += iXml.iXmlNodeSqlTableprivOpen();
-    }
+  tablePriv(params, options = {}) {
+    this.xml += `<tablepriv${options.error ? ` error='${options.error}'` : ''}>`;
 
     for (let i = 0; i < params.length; i += 1) {
-      this.xml += iXml.iXmlNodeSqlParmOpen() + params[i] + iXml.iXmlNodeSqlParmClose();
+      this.xml += `<parm>${params[i]}</parm>`;
     }
 
-    this.xml += iXml.iXmlNodeSqlTableprivClose();
+    this.xml += '</tablepriv>';
   }
 
   /**
    * @description adds sql columns XML
    * @param {array} params
    * @param {object} [options]
    */
-  columns(params, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlColumnsOpen(options.error);
-    } else {
-      this.xml += iXml.iXmlNodeSqlColumnsOpen();
-    }
+  columns(params, options = {}) {
+    this.xml += `<columns${options.error ? ` error='${options.error}'` : ''}>`;
 
     for (let i = 0; i < params.length; i += 1) {
-      this.xml += iXml.iXmlNodeSqlParmOpen() + params[i] + iXml.iXmlNodeSqlParmClose();
+      this.xml += `<parm>${params[i]}</parm>`;
     }
-    this.xml += iXml.iXmlNodeSqlColumnsClose();
+    this.xml += '</columns>';
   }
 
   /**
    * @description adds sql special XML
    * @param {array} params
    * @param {object} [options]
    */
-  special(params, options) {
-    if (options && options.error) {
-      this.xml += iXml.iXmlNodeSqlSpecialOpen(options.error);
-    } else {
-      this.xml += iXml.iXmlNodeSqlSpecialOpen();
-    }
+  special(params, options = {}) {
+    this.xml += `<special${options.error ? ` error='${options.error}'` : ''}>`;
```

Comment:
Looks like there's a bunch of them. All of them should be adjusted.

Reply:
All should be fixed now